x-pack/plugins/ml/server/models/model_management/models_provider.ts

```diff
@@ -581,4 +585,37 @@ export class ModelsProvider {
     await mlSavedObjectService.updateTrainedModelsSpaces([modelId], ['*'], []);
     return putResponse;
   }
+  /**
+   * Puts the requested Inference endpoint id into elasticsearch, triggering elasticsearch to create the inference endpoint id
+   * @param inferenceId
+   * @param taskType
+   * @param modelConfig
+   */
+  async createInferenceEndpoint(
+    inferenceId: string,
+    taskType: InferenceTaskType,
+    modelConfig: InferenceModelConfig
+  ) {
+    let esModelExists = false;
+    try {
+      await this._client.asInternalUser.inference.getModel({ inference_id: inferenceId });
+
+      esModelExists = true;
+    } catch (error) {
+      if (error.statusCode !== 404) {
+        throw error;
+      }
+      // model doesn't exist, ignore error
+    }
+
+    if (esModelExists) {
+      throw Boom.badRequest('Model already exists');
+    }
```

Comment:
do you need this logic? doesn't the ES endpoint (this._client.asCurrentUser.inference.putModel) take care of the validation?

Reply:
Good catch. This logic is not needed. 
Fixed in https://github.com/elastic/kibana/pull/180330/commits/c6138c66cd8ac2ec7daf3283a8962e6d49c15911